lib/swaps/Swaps.ts

```diff
@@ -681,12 +693,16 @@ class Swaps extends EventEmitter {
     let expectedAmount: number;
     let source: string;
     let destination: string;
-
+    // TODO: check cltv value
     switch (deal.role) {
       case SwapRole.Maker:
         expectedAmount = deal.makerAmount;
         source = 'Taker';
         destination = 'Maker';
+        if (deal.makerCltvDelta! > 50 * 2) {
+          this.failDeal(deal, SwapFailureReason.UnknownError, 'Wrong CLTV received on first leg');
```

Comment:
Maybe use `SwapFailureReason.InvalidResolveRequest` instead of the unknown error?

Reply:
Agree